### PR TITLE
feat(ext.pages): add more ways to send paginator by editing

### LINKED ISSUE #1877: Paginator View doesn't respond if paginator buttons were added via editing the message
### Summary

This seems to be happening with Paginator.edit() in all cases ive been able to test

### Reproduction Steps

Send a message using e.g. ctx.respond or ctx.send
create a paginator object 
and use await `paginator.edit(message)` to add the paginator view to the message
The buttons appear, but they dont work, this also occurs when using a previous view in the initial message which items from that become unresponsive too.

### Minimal Reproducible Code

```python
async def test_cmd(self, ctx: bridge.BridgeContext):
    embed = utils.minimal_embed("Hello World")
    message = await ctx.respond(embed=embed)
    if isinstance(message, discord.Interaction):
        message = await message.original_response()
    if isinstance(message, (discord.WebhookMessage, discord.PartialMessage)):
        message = self.client.get_message(message.id)
    paginator = pages.Paginator(pages=["message 1", "message 2", "message 3", "message 4"])
    await paginator.edit(message)
```

### Expected Results

The buttons should respond like other views when they are edited, they work fine

### Actual Results

Instead I get a "This Interaction Failed" when clicking any of the buttons

### Intents

all

### System Information

discord/__main__.py:47: DeprecationWarning: VersionInfo.releaselevel is deprecated since version 2.3, consider using release_level instead.
  "- py-cord v{0.major}.{0.minor}.{0.micro}-{0.releaselevel}".format(version_info)
discord/__main__.py:49: DeprecationWarning: VersionInfo.releaselevel is deprecated since version 2.3, consider using release_level instead.
  if version_info.releaselevel != "final":
- Python v3.10.9-final
- py-cord v2.3.2-final
- aiohttp v3.8.3
- system info: Linux 6.1.6-arch1-1 #&#8203;1 SMP PREEMPT_DYNAMIC Sat, 14 Jan 2023 13:09:35 +0000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Om1609**
## Summary

Allows the Paginator to be sent using the edit methods offered by `Interaction`, `ApplicationContext`, and `BridgeContext`.

Also fixes #1877 by adding a new parameter `user`, when passing a `Message` object. 

`paginator.edit()` didn't work at all, because `paginator.user` was being set to `message.author`. But `message.author` will always be the bot itself, since it can edit that message. This caused the views to not respond, unless `author_check` was disabled. This means a user would need be to be explicitly accepted while calling `paginator.edit()`, thus the new parameter.

## Information

- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by Om1609** [comment]
On hold. There is parameter discrepancy between methods. 

`Interaction.edit_original_response` doesn't accept `suppress` kwarg, since discord doesn't support it for webhook message edits.

`InteractionResponse.edit_message` doesn't seem to accept `suppress` and `allowed_mention` kwargs, even though it looks like discord accepts it in the flags ([link](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-messages)).

**Comment by Om1609** [comment]
Should be good after #2138 is merged

**Review by Lulalaby — APPROVED** [review]

File: CHANGELOG.md
Comment (Dorukyum) [inline CHANGELOG.md:74]: Not sure if this change counts as an addition instead of a change.

Comment (Lulalaby) [reply CHANGELOG.md:74]: I think added is fine

Comment (Om1609) [reply CHANGELOG.md:74]: I could change the line to `Added more classes in paginator.edit.` if that makes it better.

Comment (Lulalaby) [reply CHANGELOG.md:74]: yeh

**Review by Lulalaby — APPROVED** [review]

File: discord/ext/pages/pagination.py
Comment (Om1609) [inline discord/ext/pages/pagination.py:1101]: Should this be `discord.ext.bridge.BridgeContext` or `discord.ext.bridge.Context`? Or even `discord.ext.bridge. BridgeApplicationContext` and `discord.ext.bridge. BridgeExtContext` individually?

**Review by JustaSqu1d — APPROVED** [review]

**Comment by Om1609** [comment]
Yk what, i'll make an alias for the parameter, so there is no breaking change

File: discord/ext/pages/pagination.py
Comment (JustaSqu1d) [inline discord/ext/pages/pagination.py:1102]: Clarify what we mean here by "target"?

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: discord/ext/pages/pagination.py
Comment (Om1609) [reply discord/ext/pages/pagination.py:1102]: Now that I think of it, `message` actually made more sense. All these methods are editing a "message" in a channel. I change this back to `message`

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
